### PR TITLE
adds option to pick number of availability zones when creating clusters

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/pull/2202. 

screen shot 2018-11-27 at 11 48 10

screen shot 2018-11-27 at 11 48 19

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "connected-react-router": "4.5.0",
     "copy-to-clipboard": "~3.0.8",
     "deep-diff": "1.0.2",
-    "giantswarm-v4": "git+https://github.com/giantswarm/giantswarm-js-client.git#master",
+    "giantswarm-v4": "git+https://github.com/giantswarm/giantswarm-js-client.git#availability-zones",
```
Comment (xh3b4sd) [inline package.json]: See https://github.com/giantswarm/giantswarm-js-client/pull/56.

**Comment by xh3b4sd** [comment]
How to solve this? It somehow does not tell me what is actually wrong. 

```
$ make validate-prettier
docker run --rm -ti \
		-v /Users/xh3b4sd/projects/giantswarm/happa/src:/usr/src/app/src \
		-v /Users/xh3b4sd/projects/giantswarm/happa/dist:/usr/src/app/dist \
		-v /Users/xh3b4sd/projects/giantswarm/happa/node_modules_linux:/usr/src/app/node_modules \
		-v /Users/xh3b4sd/projects/giantswarm/happa/package.json:/usr/src/app/package.json \
		-v /Users/xh3b4sd/projects/giantswarm/happa/yarn.lock:/usr/src/app/yarn.lock \
		quay.io/giantswarm/happa-build:latest yarn run validate-prettier
yarn run v1.10.1
$ prettier --jsx-single-quote --single-quote --trailing-comma es5 --list-different "src/**/*.js"
/bin/sh: 1: prettier: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [validate-prettier] Error 1
```

**Comment by oponder** [comment]
You gotta run prettier on your code base for the code to pass. Somehow though your local run is complaining that the `prettier` binary is not found: `prettier: not found`

That is weird. Try running `make install-node-modules` first.

`make validate-prettier` won't give you any diff though, it just fails or passes.

To apply prettier to all your files you can run: 

```
	docker run --rm -ti \
		-v ${PWD}/src:/usr/src/app/src \
		-v ${PWD}/dist:/usr/src/app/dist \
		-v ${PWD}/node_modules_linux:/usr/src/app/node_modules \
		-v ${PWD}/package.json:/usr/src/app/package.json \
		-v ${PWD}/yarn.lock:/usr/src/app/yarn.lock \
		quay.io/giantswarm/happa-build:latest yarn run prettier
```

**Comment by marians** [comment]
I'd love to test a deployed version. Please ping me when there is one.

**Comment by xh3b4sd** [comment]
Oli and I paired a bit more and made the thing work for other providers and other release versions. Please have a look again and provide some suggestions on wording and so on.

**Comment by xh3b4sd** [comment]
@marians you can test the current version on `ginger` if you like.

**Comment by marians** [comment]
Note to future happa coders: when the constructor is only there to initialize the state, like here

```javascript
class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {...};
  }
}
```

that can be simplified like this:

```javascript
class MyComponent extends React.Component {
  state = {...};
}
```

File: src/components/create_cluster/index.js [hunk 2]
```diff
@@ -316,6 +326,56 @@ class CreateCluster extends React.Component {
...
+                        </p>
+                        <div className='col-3'>
+                          <NumberPicker value={1} readOnly={true} />
+                        </div>
+                      </div>
+                    )
+                  ) : (
+                    <p>
+                      Selection of availability zones on this installation is
+                      limited to 1 availability zone.
```
[the remark this answers is not in the document]
Comment (xh3b4sd) [reply src/components/create_cluster/index.js]: Whatever you prefer. Done.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

**Comment by marians** [comment]
And some  🎉 🚀 🍾  for this PR :)

**Comment by oponder** [comment]
kudos tim!

**Pull request merged**